buildsys: Fix `make distclean`

k-takata opened the pull request:
We use Automake, so we can use `make dist` to create a tarball and `make distcheck` to create and check it.
However, `make distcheck` fails because of some reasons.

* Source files are not properly listed. We also need to care about Windows specific source files.
* Creating a tarball fails because of very long filenames.
  See: https://www.gnu.org/software/automake/manual/html_node/List-of-Automake-options.html
  `tar-v7` supports file names with up to only 99 characters.
  However we have some files which names are longer than that.
  So we should use `tar-ustar` or `tar-pax`.
* `make distclean` leaves some files in build directory. So the following error will be shown:
  ```
  ERROR: files left in build directory after distclean:
  ./Units/conflib-recursive.d/BUNDLES
  ./Units/conflib-recursive.d/optlib/lang.d/def.d/a.ctags
  ...
  ```
  The outputs of unit tests should be cleaned by `make clean` or `make distclean`.

k-takata commented:
Note that this PR only fixes the error of `make distcheck`, however there are still some problems with `make dist`. Documents and some files (especially for VC) are lacking from the tarball.

masatake commented:
I don't object but I don't think we have to add files under docs directory to the release tarball.
Instead I would like to add files under man directory to the tarball when we update the man page.

k-takata commented:
Updated.
Dotfiles and Win32 files are added.
Maybe some files under `misc` should be also added.

> I don't think we have to add files under docs directory to the release tarball.

Do you consider the files under `docs` are mainly for the Web page and no need to be installed in a user's PC?
Some software install HTML document under `/usr/share/doc/` (or somewhere else).

masatake commented:
> Do you consider the files under docs are mainly for the Web page and no need to be installed in a user's PC?
> Some software install HTML document under /usr/share/doc/ (or somewhere else).

The docs directory is a kind of staging area.

* Advertising experimental feature. I would like to get feed back from advanced users before releasing.
* Publishing the latest information (news) about u-ctags development. "latest" is important than its quality
  of contents. As you know I'm bad in English.
* Encouraging potential contributors. As I did in  #1425.

I think average users may need only some part of above.

The man page should 

* explain only stable features and interface,
* be written in good? English, and
* be written for average users, not for hackers.

In other word docs files are stocked materials for writing the real main page. Some parts of the files can be used and the other will be dropped.

I will not object to including docs files. However, at least, I would like you to know my intention.

We can add another document like INSTALL, NEWS, ... We have to think about this. However, for awhile, I would like to think about updating ctags.1 (, ctags-format.1, ctags-incompat.1, and...).

Do I make sense?

k-takata commented:
Thanks. I got your intention.
I added html files generated from docs/*.rst in Windows package with #1410, but then is it better to add an html file generated from man/ctags.1.rst?

Anyway I will merge this PR for now. `EXTRA_DIST` should be kept maintained.

masatake commented:
> I added html files generated from docs/*.rst in Windows package with #1410, but then is it better to add an html file generated from man/ctags.1.rst?

Yes. man/ctags.1.rst is must be included. docs/*.rst (or html) are optional.
I'm thinking about splitting man/ctags.1.rst into some.
